src/app.tsx

```diff
@@ -1,15 +1,30 @@
 declare let window: any;
-import React, { FunctionComponent } from "react";
+import React, { FunctionComponent, useEffect, useState } from "react";
 import "./app.css";
 import { MainPage } from "./page/main-page";
 import { MetaMaskError } from "./page/metamask-error";
+import { MetaMaskPending } from "./page/metamask-pending";
 
 const App: FunctionComponent = () => {
+  const [metamaskConnected, setMetamaskConnected] = useState(false);
+  useEffect(() => {
+    checkMetaMaskAccount();
+  });
+
+  const checkMetaMaskAccount = async () => {
+    try {
+      const accounts = await window.ethereum.request({ method: "eth_requestAccounts" });
+      accounts.length > 0 ? setMetamaskConnected(true) : setMetamaskConnected(false);
+    } catch (e) {
+      e.code === 4001 ? setMetamaskConnected(true) : console.debug(e);
```

Comment:
is this documented somewhere ? would be interesting to link to.

Reply:
https://docs.metamask.io/guide/ethereum-provider.html#using-the-provider.
 I read the documentation from here and did some modifications.
 i will add a comment at the top to display the link

Reply:
https://docs.metamask.io/guide/ethereum-provider.html#errors

it's documented here. can add the link into the code ?

Reply:
done.